cub/cub/thread/thread_operators.cuh

```diff
@@ -634,82 +583,160 @@ struct SimdMul<__half>
 template <>
 struct SimdMul<__nv_bfloat16>
 {
-  using simd_type = __nv_bfloat162;
-
-  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __nv_bfloat162 operator()(__nv_bfloat162 a, __nv_bfloat162 b) const
+  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE __nv_bfloat162
+  operator()([[maybe_unused]] __nv_bfloat162 a, [[maybe_unused]] __nv_bfloat162 b) const
   {
-#    if _CCCL_CUDACC_BELOW(12) && _CCCL_CUDA_COMPILER(NVHPC)
-    return __floats2bfloat162_rn(
-      __bfloat162float(a.x) * __bfloat162float(b.x), __bfloat162float(a.y) * __bfloat162float(b.y));
-#    else // ^^^ _CCCL_CUDACC_BELOW(12) && _CCCL_CUDA_COMPILER(NVHPC) ^^^ / vvv otherwise vvv
     NV_IF_TARGET(NV_PROVIDES_SM_80,
-                 (return __hmul2(a, b);),
-                 (return halves2bfloat162(__float2bfloat16(__bfloat162float(a.x) * __bfloat162float(b.x)),
-                                          __float2bfloat16(__bfloat162float(a.y) * __bfloat162float(b.y)));));
-#    endif // !_CCCL_CUDACC_BELOW(12) || !_CCCL_CUDA_COMPILER(NVHPC)
+                 (return __hmul2(a, b);), //
+                 (return simd_operation_is_not_supported_before_sm53(__nv_bfloat162{});));
   }
 };
 
 #  endif // _CCCL_HAS_NVBF16()
 
 //----------------------------------------------------------------------------------------------------------------------
+// Predefined operators
+
+using cub::detail::is_one_of_v;
```

Comment:
Nitpick, this is only used 3 times, I would rather not pull it into the whole namespace for 3 uses

Reply:
We may see something like that C++29: https://wg21.link/P3625 (will be in the post-Hagenberg mailing)